src/components/ContactCard/ContactFormField.jsx

```diff
@@ -75,49 +92,90 @@ ContactFieldInput.defaultProps = {
   labelPlaceholder: ""
 };
 
-const ContactFormField = ({
-  icon,
-  name,
-  type,
-  label,
-  placeholder,
-  inputWithLabel,
-  required,
-  labelPlaceholder
-}) => (
-  <div className="contact-form__field">
-    <label className="contact-form__label">
-      {icon && (
-        <Icon icon={icon} color="coolGrey" className="contact-form__icon" />
-      )}
-      {label}
-    </label>
-    <ContactFieldInput
-      name={name}
-      type={type}
-      placeholder={placeholder}
-      withLabel={inputWithLabel}
-      required={required}
-      labelPlaceholder={labelPlaceholder}
-    />
-  </div>
-);
+export class ContactFormField extends React.Component {
+  render() {
+    const { name, icon, label, isArray, renderInput } = this.props;
+    return (
+      <div className="contact-form__field">
+        <label className="contact-form__label">
+          {icon && (
+            <Icon icon={icon} color="coolGrey" className="contact-form__icon" />
+          )}
+          {label}
+        </label>
+        {isArray ? (
+          <FieldArray name={name}>
+            {({ fields }) => (
+              <div className="contact-form__inputs-wrapper">
```

Comment:
[Not blocking] à vue de nez, on ne pourrait pas le mettre en parent de FieldArray plutôt que l'inverse ? ça permettrait de le mutualiser avec la L123 https://github.com/cozy/cozy-contacts/pull/78/files#diff-7fe9fa8d96002866bee0233c53e45554R123

Reply:
Malheureusement non.
Si on enlève ce `div` ici, l'enfant-fonction de `FieldArray` va renvoyer un tableau de composants (le résultat de `fields.map`). Mais avoir un composant qui renvois un tableau de composant est une nouvelle feature de react 16, que preact ne supporte pas (encore). Donc dans notre cas, il faut renvoyer un seul composant… ici le `div` wrapper.